server/src/main/java/org/elasticsearch/cluster/SnapshotDeletionsInProgress.java

```diff
@@ -176,17 +231,23 @@ public String toString() {
     public static final class Entry implements Writeable, RepositoryOperation {
         private final List<SnapshotId> snapshots;
         private final String repoName;
+        private final State state;
```

Comment:
Needed to know whether or not a delete can be executed yet or is still open to have snapshot ids added to it. Somewhat analogous to snapshot in progress state.
Technically, a boolean would be enough here. But I'd like to keep that state enum here to maybe further enhance the state machine with a cleanup step. As an improvement for concurrent operations, we could already move on to the next snapshot operation after master is done writing the new metadata during a delete but before it deleted all the unreferrenced blobs. It would be nice for the cleanup to still be recorded in the cluster state though as otherwise back-to-back deletes might list out and delete the same blobs concurrently+redundantly. It would just be easier to create that follow-up if we didn't go with a flag here now.